client/extensions/woocommerce/state/sites/settings/products/actions.js

```diff
@@ -79,3 +115,29 @@ export const updateSettingsProducts = ( siteId, settingsData ) => dispatch => {
 			} );
 		} );
 };
+
+/**
+ * Updates settings/products via batch endpoint.
+ *
+ * @param {Number} siteId wpcom site id.
+ * @param {Object}  successAction, success action object
+ * @param {Object}  failureAction, failure action object
+ * @return {Object} Action object
+ */
+export const saveWeightAndDimensionsUnits = ( siteId, successAction, failureAction ) => (
+	dispatch,
+	getState
+) => {
+	if (
+		! areSettingsProductsLoaded( getState(), siteId ) ||
```

Comment:
Since `getState()` is called multiple times, would it make sense to just call it once and assign it to a `const` in this scope?

Also curious as to why the check to see if settings are loaded here? Are there instances where this could even happen via the UI and/or does it matter if the settings have been loaded?

Reply:
`getState()` cached

No there is no chance for this to happen via UI. I just thought it would be good to check if we are later fetching some values which make sense only when settings are loaded.